polars/polars-time/src/chunkedarray/utf8/strptime.rs

```diff
@@ -50,10 +50,17 @@ pub(super) fn compile_fmt(fmt: &str) -> String {
 // # Safety
 // Caller must ensure that fmt adheres to the fmt rules of chrono and `fmt_len` is correct.
 pub(super) unsafe fn parse(val: &[u8], fmt: &[u8], fmt_len: u16) -> Option<NaiveDateTime> {
-    const ESCAPE: u8 = b'%';
-    if val.len() < fmt_len as usize {
+    let mut offset = 0;
+    let mut negative = false;
+    if val.starts_with(b"-") && fmt.starts_with(b"%Y") {
+        offset = 1;
+        negative = true;
+    }
+    if val.len() - offset != (fmt_len as usize) {
```

Comment:
having removed `'%Z'`, all other directives are fixed-length, so this check can be made more precise